### DISCUSSION TITLE
支持「术语表」或「用户词典」

### DISCUSSION THREAD

**Discussion by Lu-233** [immersive-translate/immersive-translate#1689] [Ideas]
### 新功能 | New Feature

支持用户自定义词典，如DeepL支持的「术语表」。
用户可以为全局或网站设置一些中英词汇对照。
这一特性可以在：支持使用术语表的翻译服务，支持自定义prompt的GPT服务中使用。
这些术语表可以共享，类似UGC的模式可以提升所有用户的体验。

### 为什么您想要添加这些功能 | Why you want to add these features

很多场景下，机翻不会顾及领域知识和上下文。
比如各领域的专业文献，新闻甚至游戏帖子中，都需要一些术语对照来提升翻译效果。错误的术语翻译可能导致整个句子翻译错误。

### 提议 | Proposal

利用机翻API对「术语表」的支持，和GPT自定义的prompt实现功能。
利用已有的分享渠道和官方网站来分享用户自定义词典。这种机制也可以分享自定义的CSS。

**Comment by theowenyoung** [comment]
目前可以在 【进阶设置】里设置一些不需要翻译的词语。

更多的术语表 在客户端做效果不太好。
